pkgs/network/src/mock.zig

```diff
@@ -55,3 +55,94 @@ pub const Mock = struct {
         } };
     }
 };
+
+// Helper functions and structs for testing
+const SubscriberData = struct {
+    calls: *u32,
```

Comment:
1. declare these types inside test
2. why do calls and received pointers, this way you won't have to assign them them from outside passing references
3. onGossip should be a part of this subscriber itself (see how node class/struct does)
4. once you create a mock object, get network = mock.getNetworkInterface() or something and then you should do network.subscribe so that that entire flow is covered in the testcase, again see node class